llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -813,6 +813,8 @@ void llvm::computeKnownBitsFromContext(const Value *V, KnownBits &Known,
     if (!isValidAssumeForContext(I, Q.CxtI, Q.DT))
       continue;
 
+    // Change to `computeKnownBitsFromCond` if we start supporting assume of
+    // logic op in ConditionCacheUtil.
```

Comment:
Conditions for `@llvm.assume` should be atomic expressions. TBH I don't think we can benefit from handling logical ops here.

Reply:
Something like `assume(a || b)` can exist.

I'm happy to drop supporting it in this patch, but would prefer to keep comments indicating the steps necessary should we change out mind.

that okay?

Reply:
> Something like `assume(a || b)` can exist.

Yeah, but it would not benefit the optimization if we fail to merge `a || b`.

Reply:
What do you mean we fail to merge the `a || b`?

Reply:
Due to recent unfortunate events, I would recommend against leaving any TODO/FIXME comments unless you are very sure that resolving them is actually valuable. Otherwise there is too much risk that new contributors will try to "fix" them without having the necessary background.